File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        table = [0] * 26
        for c in s:
            table[ord(c) - ord('a')] += 1
        for c in t:
            table[ord(c) - ord('a')] -= 1
        for n in table:
            if n != 0:
                return False       
        return True

File: 0242-valid-anagram/0242-valid-anagram.cpp
// vector or unordered_map to count
class Solution {
public:
    bool isAnagram(string s, string t) {

        if (s.length() != t.length())
            return false;

        vector<int> cnts(26);
        int n = s.length();
        for (int i = 0; i < n; i++) {
            cnts[s[i] - 'a']++;
            cnts[t[i] - 'a']--;
        }

        for (int i = 0; i < cnts.size(); i++) {
            if (cnts[i] != 0)
                return false;
        }
        return true;
    }
};